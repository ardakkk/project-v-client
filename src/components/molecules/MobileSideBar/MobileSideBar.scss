.MobileSideBar {
    width: 'auto';

    &_content {
        width: 270px;
    }

    &_link {
        color: inherit;
        text-decoration: none;
    }
}
