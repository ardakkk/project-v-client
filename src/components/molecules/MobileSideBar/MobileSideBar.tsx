import * as React from 'react';

import Divider from '@material-ui/core/Divider';
import Drawer from '@material-ui/core/Drawer';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import AddAlarm from '@material-ui/icons/AddAlarm';
import Book from '@material-ui/icons/Book';
import CalendarToday from '@material-ui/icons/CalendarToday';
import DateRange from '@material-ui/icons/DateRange';
import GpsFixed from '@material-ui/icons/GpsFixed';
import InboxIcon from '@material-ui/icons/MoveToInbox';

import * as classes from './MobileSideBar.scss';

interface IMobileSideBarProps {
    isOpenLeftSideBar: boolean;
    toggleLeftSideBarDrawer: (event: any) => void;
}

export const MobileSideBar: React.FunctionComponent<IMobileSideBarProps> = ({
    isOpenLeftSideBar,
    toggleLeftSideBarDrawer,
}) => {
    return (
        <div className={classes.MobileSideBar}>
            <Drawer open={isOpenLeftSideBar} onClose={toggleLeftSideBarDrawer}>
                <div
                    className={classes.MobileSideBar_content}
                    role='presentation'
                    onClick={toggleLeftSideBarDrawer}>
                    <List>
                        <ListItem button={true}>
                            <ListItemIcon>
                                <InboxIcon />
                            </ListItemIcon>
                            <ListItemText primary='Icebox' />
                        </ListItem>
                        <ListItem button={true}>
                            <ListItemIcon>
                                <CalendarToday />
                            </ListItemIcon>
                            <ListItemText primary='Today' />
                        </ListItem>
                        <ListItem button={true}>
                            <ListItemIcon>
                                <DateRange />
                            </ListItemIcon>
                            <ListItemText primary='Next week' />
                        </ListItem>
                    </List>
                    <Divider />
                    <List>
                        <ListItem button={true}>
                            <ListItemIcon>
                                <GpsFixed />
                            </ListItemIcon>
                            <ListItemText primary='Goals' />
                        </ListItem>
                        <ListItem button={true}>
                            <ListItemIcon>
                                <AddAlarm />
                            </ListItemIcon>
                            <ListItemText primary='Alarm' />
                        </ListItem>
                        <ListItem button={true}>
                            <ListItemIcon>
                                <Book />
                            </ListItemIcon>
                            <ListItemText primary='Diary' />
                        </ListItem>
                    </List>
                </div>
            </Drawer>
        </div>
    );
};
