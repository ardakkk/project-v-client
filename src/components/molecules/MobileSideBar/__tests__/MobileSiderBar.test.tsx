import { shallow } from 'enzyme';
import * as React from 'react';

import { MobileSideBar } from '../MobileSideBar';

describe('MobileSideBar component.', () => {
    it('Snapshot should match.', () => {
        const component = shallow(
            <MobileSideBar
                isOpenLeftSideBar={true}
                toggleLeftSideBarDrawer={() => {}}
            />
        );

        expect(component).toMatchSnapshot();
    });
});
