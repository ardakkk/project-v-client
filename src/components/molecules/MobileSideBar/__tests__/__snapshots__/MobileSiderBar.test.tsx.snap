// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileSideBar component. Snapshot should match. 1`] = `
<div>
  <WithStyles(ForwardRef(Drawer))
    onClose={[Function]}
    open={true}
  >
    <div
      onClick={[Function]}
      role="presentation"
    >
      <WithStyles(ForwardRef(List))>
        <Link
          to="/icebox"
        >
          <WithStyles(ForwardRef(ListItem))
            button={true}
          >
            <WithStyles(ForwardRef(ListItemIcon))>
              <MoveToInboxIcon />
            </WithStyles(ForwardRef(ListItemIcon))>
            <WithStyles(ForwardRef(ListItemText))
              primary="Icebox"
            />
          </WithStyles(ForwardRef(ListItem))>
        </Link>
        <Link
          to="/"
        >
          <WithStyles(ForwardRef(ListItem))
            button={true}
          >
            <WithStyles(ForwardRef(ListItemIcon))>
              <CalendarTodayIcon />
            </WithStyles(ForwardRef(ListItemIcon))>
            <WithStyles(ForwardRef(ListItemText))
              primary="Today"
            />
          </WithStyles(ForwardRef(ListItem))>
        </Link>
        <Link
          to="/next-week"
        >
          <WithStyles(ForwardRef(ListItem))
            button={true}
          >
            <WithStyles(ForwardRef(ListItemIcon))>
              <DateRangeIcon />
            </WithStyles(ForwardRef(ListItemIcon))>
            <WithStyles(ForwardRef(ListItemText))
              primary="Next week"
            />
          </WithStyles(ForwardRef(ListItem))>
        </Link>
      </WithStyles(ForwardRef(List))>
      <WithStyles(ForwardRef(Divider)) />
      <WithStyles(ForwardRef(List))>
        <WithStyles(ForwardRef(ListItem))
          button={true}
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <GpsFixedIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Goals"
          />
        </WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItem))
          button={true}
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <AddAlarmIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Alarm"
          />
        </WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItem))
          button={true}
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <BookIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Diary"
          />
        </WithStyles(ForwardRef(ListItem))>
      </WithStyles(ForwardRef(List))>
    </div>
  </WithStyles(ForwardRef(Drawer))>
</div>
`;
